core/domain/takeout_commit_message_truncate_jobs_one_off.py

```diff
@@ -0,0 +1,132 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""One-off jobs for suggestions."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core import jobs
+from core.platform import models
+
+(
+    base_models, config_models, collection_models, exploration_models,
+    skill_models, story_models, topic_models, subtopic_models,
+    question_models
+) = models.Registry.import_models([
+    models.NAMES.base_model, models.NAMES.config, models.NAMES.collection,
+    models.NAMES.exploration, models.NAMES.skill, models.NAMES.story,
+    models.NAMES.topic, models.NAMES.subtopic, models.NAMES.question
+])
+
+
+class SnapshotMetadataCommitMsgAuditOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """Job that audits the commit_message field of the
+    BaseSnapshotMetadataModels to determine frequency of string sizes. This job
+    is temporary and only used once.
+    """
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        super(SnapshotMetadataCommitMsgAuditOneOffJob, cls).enqueue(
+            job_id, shard_count=64)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [
+            config_models.ConfigPropertySnapshotMetadataModel,
+            config_models.PlatformParameterSnapshotMetadataModel,
+            collection_models.CollectionSnapshotMetadataModel,
```

Comment:
Should also `collection_models.CollectionRightsSnapshotMetadataModel` be added? Also, what about the `commit_message` in commit log models?

Reply:
added it. I'm only doing this for models that derive from the BaseSnapshotMetadataModel, since this is where we do the projection query.

Reply:
So the `commit_message` in the commit log models was not updated even before when we were doing some modifications to them (in snapshots)? If, not could you create an issue that we should synchronize the `commit_message` in snapshots and in commit logs?

Reply:
I don’t think I fully understand your question. Are you saying the commit log models should be synchronized with the BaseSnapshotMetadataModel commit messages?

Reply:
Yep, I think so, it might be unnecessary to keep both in the future, but for now let's make them the same in all models. I'm fine if you did this in another PR.

Reply:
@vojtechjelinek Sounds good. I'll create an issue for this.